File: Problems & Solutions/1470. Shuffle the Array/06 February 2023/Approach 2 - In-place Filling/Shuffle the Array.java
class Solution {
    public int[] shuffle(int[] nums, int n) {
        for (int i = n; i < 2 * n; ++i) {
            int secondNum = nums[i] << 10;
            nums[i - n] |= secondNum;
        }

        int allOnes = (int) Math.pow(2, 10) - 1;

        for (int i = n - 1; i >= 0; --i) {
            int secondNum = nums[i] >> 10;
            int firstNum = nums[i] & allOnes;
            nums[2 * i + 1] = secondNum;
            nums[2 * i] = firstNum;
        }
        return nums;
    }
}

File: Problems & Solutions/1470. Shuffle the Array/06 February 2023/Approach 2 - In-place Filling/Shuffle the Array.js
var shuffle = function(nums, n) {
    for (let i = n; i < 2 * n; ++i) {
        let secondNum = nums[i] << 10;
        nums[i - n] |= secondNum;
    }

    let allOnes = Number(Math.pow(2, 10)) - 1;

    for (let i = n - 1; i >= 0; --i) {
        let secondNum = nums[i] >> 10;
        let firstNum = nums[i] & allOnes;
        nums[2 * i + 1] = secondNum;
        nums[2 * i] = firstNum;
    }
    return nums;
};